torch/_functorch/aot_autograd.py

```diff
@@ -1052,12 +1062,20 @@ def inner(*flat_args):
 
         flat_f_args = pytree.tree_map(_to_fun, flat_args)
 
+        prior_autocast_states = _get_autocast_states()
+
         # See Note [Disabling Functionalize TLS Above Python Functionalization]
         disable_above = torch._C._ExcludeDispatchKeyGuard(torch._C.DispatchKeySet(torch._C.DispatchKey.Functionalize))
         with disable_above, FunctionalTensorMode():
             # precondition: The passed in function already handles unflattening inputs + flattening outputs
             flat_f_outs = f(*flat_f_args)
 
+        if prior_autocast_states != _get_autocast_states():
+            raise RuntimeError(
+                "AOTAutograd does not support tracing graphs that mutate "
+                "the autocast state"
```

Comment:
Nit: can you improve the error message here? Something like:

"AOTAutograd does not support tracing graphs that mutate the autocast state. When run with torch.compile, we expect dynamo to graph break on calls to `torch.set_autocast_enabled()`. When using torch.compile, you should prefer calls to the autocast context manager over this direct API."

Reply:
Technically though, this isn't a problem on dynamo since if we graph break, it never makes it into the graph,

Reply:
I encourage users to file a bug if they encounter this error as it should never happen when using torch.compile